### PR TITLE
feat/ED-2062-no-json-encoding-for-xml-mimetypes

### PR DESCRIPTION
**Pull request by jkumara**
Handle requests with XML mimetypes correctly. I did a check on XML mimetype rather than on JSON mimetype to minimize the change of accidentally breaking something😅

### CODE REVIEW

File: src/gen/js/service.js.template
Comment (villelahdenvuo) [inline src/gen/js/service.js.template:1]: Must also change in the ts template?

**Review by villelahdenvuo — CHANGES_REQUESTED** [review]
Change needed in TS template?

File: src/gen/js/service.js.template
Comment (jkumara) [reply src/gen/js/service.js.template:1]: Yes, I missed that.

File: src/gen/js/service.ts.template [hunk 1]
```diff
@@ -65,7 +65,9 @@ function acquireRights(
 }
 
 function makeFetchRequest(req: api.ServiceRequest): Promise<Response> {
-	let fetchOptions: any = {
+  const isXmlRequest = req.headers['Content-Type']?.includes('/xml');
+  const body = isXmlRequest ? req.body : JSON.stringify(req.body || undefined);
+  const fetchOptions = {
```
Comment (villelahdenvuo) [inline src/gen/js/service.ts.template]: indent

File: src/gen/js/service.ts.template [hunk 2]
```diff
@@ -65,7 +65,9 @@ function acquireRights(
 }
 
 function makeFetchRequest(req: api.ServiceRequest): Promise<Response> {
-	let fetchOptions: any = {
+	const isXmlRequest = req.headers['Content-Type']?.includes('/xml');
+	const body = isXmlRequest ? req.body : JSON.stringify(req.body || undefined);
```
Comment (villelahdenvuo) [inline src/gen/js/service.ts.template:69]: Not actually used?

**Review by villelahdenvuo — CHANGES_REQUESTED** [review]

File: src/gen/js/service.ts.template (hunk 2 above)
Comment (jkumara) [reply src/gen/js/service.ts.template:69]: Argh, too hasty copy-paste

**Pull request merged**